packages/eslint-plugin-template/tests/rules/no-any.test.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  convertAnnotatedSourceToFailureCase,
+  RuleTester,
+} from '@angular-eslint/utils';
+import rule, { MessageIds, RULE_NAME } from '../../src/rules/no-any';
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: '@angular-eslint/template-parser',
+});
+
+const messageId: MessageIds = 'noAny';
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    `
+      {{ $any }}
+    `,
+    `
+      {{ obj.$any() }}
+    `,
+    `
+      {{ obj?.x?.y!.z!.$any() }}
+    `,
+    `
+      <a [href]="$test()">Click here</a>
+    `,
+    `
+      <button type="button" (click)="anyClick()">Click here</button>
+    `,
+    `
+      {{ $any }}
+      {{ obj?.x?.y!.z!.$any() }}
+      <a [href]="$test()">Click here</a>
+      <button type="button" (click)="anyClick()">Click here</button>
+    `,
+  ],
+  invalid: [
+    convertAnnotatedSourceToFailureCase({
+      description: 'it should fail with call expression in expression binding',
+      annotatedSource: `
+      {{ $any(framework).name }}
+          ~~~~~~~~~~~~~~~
+      `,
+      messageId,
+    }),
+    convertAnnotatedSourceToFailureCase({
+      description: 'it should fail with call expression using "this"',
+      annotatedSource: `
+      {{ this.$any(framework).name }}
+          ~~~~~~~~~~~~~~~~~~~~
+      `,
+      messageId,
+    }),
+    convertAnnotatedSourceToFailureCase({
+      description: 'it should fail with call expression in property binding',
+      annotatedSource: `
+      <a [href]="$any(getHref())">Click here</a>
+                 ~~~~~~~~~~~~~~~
+      `,
+      messageId,
+    }),
+    convertAnnotatedSourceToFailureCase({
```

Comment:
Hey @JamesHenry I was unable to make this test pass. I got this message:

> Message:
    Error line should be 4

 Do you have any clue?

Reply:
Have you tried using `parserServices.convertNodeSourceSpanToLoc()` like all the other template rules do?

Reply:
Hmm, not really. I followed the same idea of `no-call-expression` (which doesn't use this), but now you said I tested `parserServices.convertNodeSourceSpanToLoc` and I got the following result:

```ts
{
  start: { line: NaN, column: undefined },
  end: { line: NaN, column: undefined }
}

Reply:
Hmm I don't time to dig into this loads right now but I've noticed that if you wrap your middle assertion in a div, then it works.

Worth looking into the AST that the angular compiler actually produces for that source.

I added the div and fixed the alignment of the last issue by one char, that made the tests pass for me:

Reply:
Oh damn it wouldn't let me push to your branch...

Reply:
Added it as a suggestion below.

Reply:
Hey @JamesHenry sry for the delay, I've been busy these days. I think I'll be able to take a look at this tomorrow. Anyway it's really strange you couldn't push to my branch because I checked to "Allow edits and access [...]".

Reply:
Thanks @rafaelss95, for some reason looks like it worked via the github UI... 🤷‍♂️

Reply:
All green now. Ready for this to go in @rafaelss95?

I do think we need to figure out this "location when no parent element" issue, but I don't think it's specific to this rule so I'm happy to press ahead with this for now

Reply:
@JamesHenry sure, let's move on.